File: backend/src/main/java/ru/home/shop/controller/dto/DAddPresent.java
package ru.home.shop.controller.dto;

import lombok.Data;

import javax.validation.Valid;
import javax.validation.constraints.Digits;
import javax.validation.constraints.Min;
import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;
import java.math.BigDecimal;
import java.util.Collection;

@Data
public class DAddPresent {

    @NotNull
    @Size(min = 1, max = 50)
    private String name;

    @NotNull
    @Min(1)
    @Digits(integer = 5, fraction = 2)
    private BigDecimal price;

    @NotEmpty
    @Valid
    private Collection<DPresentItem> items;
}

File: backend/src/main/kotlin/io/github/artemy/osipov/shop/controller/dto/DAddPresent.kt
package io.github.artemy.osipov.shop.controller.dto

import jakarta.validation.Valid
import jakarta.validation.constraints.Digits
import jakarta.validation.constraints.Min
import jakarta.validation.constraints.NotEmpty
import jakarta.validation.constraints.NotNull
import jakarta.validation.constraints.Size
import java.math.BigDecimal

data class DAddPresent(
    @field:NotNull
    @field:Size(min = 1, max = 50)
    var name: String?,

    @field:NotNull
    @field:Min(1)
    @field:Digits(integer = 5, fraction = 2)
    var price: BigDecimal?,

    @field:NotEmpty
    @field:Valid
    var items: Collection<DPresentItem>?
)